test/test_autograd.py

```diff
@@ -5450,6 +5450,56 @@ def test_advanced_indexing_backwards_large(self, device):
         a.sum().backward()
         self.assertEqual(x.grad, torch.ones(n, 1, device=device))
 
+    @onlyCUDA
+    def test_reentrant_parent_error_on_cpu(self, device):
+        before = CudaMemoryLeakCheck.get_cuda_memory_usage()
+        t1 = torch.rand([3, 3], requires_grad=True)
+        t2 = torch.rand([3, 3], device=device, requires_grad=True)
+        t3 = torch.rand([3, 3], device=device, requires_grad=True)
+
+        # Parent graph cpu graph.
+        t4 = t1 * t1
+        t5 = TestAutograd.BackwardError.apply(t4)
+
+        # Child gpu graph (much longer than parent graph).
+        computations = []
+        computations.append(t2 * t2)
+        for i in range(10):
+            computations.append(computations[-1] * t2)
+        reentrant_root = computations[-1].sum()
+
+        class ReentrantFunc(Function):
+            @staticmethod
+            def forward(ctx, inp):
+                return inp
+
+            @staticmethod
+            def backward(ctx, grad):
+                # Reentrant backward in child will take much longer.
+                reentrant_root.backward()
+                return grad
+
+        # Parent gpu graph.
+        t6 = ReentrantFunc.apply(t3)
```

Comment:
What is the reason for the two different branches in the parent graph? I think it's fine if a CPU Node starts a GPU-only reentrant workload. Or did you wanted to explicitly avoid this case?

Reply:
Don't we need two branches in some sense to ensure the CPU thread has some work when we start a GPU task? If there is only a single branch of execution, wouldn't the CPU thread anyways have to wait for the GPU task to finish (since the next CPU task depends on the output of it)?

Reply:
Yes, my bad.